neqo-transport/src/connection/mod.rs

```diff
@@ -1462,6 +1462,10 @@ impl Connection {
             // We only allow one path during setup, so apply handshake
             // path validation to this path.
             path.borrow_mut().set_valid(now);
+        } else if self.role == Role::Client && packet.packet_type() == PacketType::Initial {
+            // If we are a client and we receive an Initial packet, we can
+            // assume that the server has validated the path.
+            path.borrow_mut().set_valid(now);
```

Comment:
This is obviously true, but what circumstances lead to a client not having a valid path at this point?  We are calling `setup_handshake_path` at the client as soon as the path is created, so this would have to involve some sort of migration during the handshake, right?

Put differently, does the test fail if you remove this block of code?